tomviz/ModuleManager.cxx

```diff
@@ -317,58 +327,73 @@ bool ModuleManager::deserialize(const pugi::xml_node& ns, const QDir& stateDir)
-    if (group==NULL || type==NULL)
-      {
-      qWarning() << "Invalid xml for Layout with id " << id;
-      continue;
-      }
-    vtkSmartPointer<vtkSMProxy> proxy;
-    proxy.TakeReference(pxm->NewProxy(group, type));
-    if (!tomviz::deserialize(proxy, node, &stateDir))
+    pugi::xml_node proxyNode = node.child("Proxy");
+    if (proxyNode)
       {
-      qWarning() << "Failed to create proxy of type: " << group << ", " << type;
-      continue;
+      pvState.append_copy(proxyNode);
+      pugi::xml_node viewSummary = pvViews.append_child("Item");
+      viewSummary.append_attribute("id").set_value(id);
+      viewSummary.append_attribute("name").set_value(
+          QString("View%1").arg(++numViews).toStdString().c_str());
       }
-    proxy->UpdateVTKObjects();
-    pxm->RegisterProxy("layouts", proxy);
-    locator->AssignProxy(id, proxy);
     }
-  for (pugi::xml_node node = ns.child("View"); node; node = node.next_sibling("View"))
+  for (pugi::xml_node node = ns.child("Layout"); node; node = node.next_sibling("Layout"))
     {
     vtkTypeUInt32 id = node.attribute("id").as_uint(0);
-    const char* group = node.attribute("xmlgroup").value();
-    const char* type = node.attribute("xmlname").value();
-    if (group==NULL || type==NULL)
-      {
-      qWarning() << "Invalid xml for View with id " << id;
-      continue;
-      }
-    vtkSmartPointer<vtkSMProxy> proxy;
-    proxy.TakeReference(pxm->NewProxy(group, type));
-    if (!tomviz::deserialize(proxy, node, &stateDir, locator.GetPointer()))
-      {
-      qWarning() << "Failed to create proxy of type: " << group << ", " << type;
-      continue;
-      }
-    proxy->UpdateVTKObjects();
-    pxm->RegisterProxy("views", proxy);
-    locator->AssignProxy(id, proxy);
-
-    if (node.attribute("active").as_int(0) == 1)
+    pugi::xml_node proxyNode = node.child("Proxy");
+    if (proxyNode)
       {
-      ActiveObjects::instance().setActiveView(vtkSMViewProxy::SafeDownCast(proxy));
+      pvState.append_copy(proxyNode);
+      pugi::xml_node layoutSummary = pvLayouts.append_child("Item");
+      layoutSummary.append_attribute("id").set_value(id);
+      layoutSummary.append_attribute("name").set_value(
+          QString("Layout%1").arg(++numLayouts).toStdString().c_str());
       }
     }
+  this->Internals->node = ns;
+  this->Internals->dir = stateDir;
+  this->connect(pqApplicationCore::instance(),
+      SIGNAL(stateLoaded(vtkPVXMLElement*,vtkSMProxyLocator*)),
+      SLOT(onPVStateLoaded(vtkPVXMLElement*,vtkSMProxyLocator*)));
+  std::ostringstream stream;
+  document.first_child().print(stream);
+  vtkNew<vtkPVXMLParser> parser;
+  if (!parser->Parse(stream.str().c_str()))
+    {
+      return false;
+    }
+  pqActiveObjects* activeObjects = &pqActiveObjects::instance();
+  pqServer *server = activeObjects->activeServer();
+
+  pqApplicationCore::instance()->loadState(parser->GetRootElement(),
+                                           server);
+  return true;
```

Comment:
You're missing a `disconnect` here. I'd also cleanup the `Internals->node` and `Internals->dir` here since they are no longer applicable once this method returns.